packages/react-components/react-tree/stories/Tree/Virtualization.stories.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {
+  TreeProps,
+  TreeItem,
+  TreeItemLayout,
+  FlatTreeItemProps,
+  useFlatTreeItems_unstable,
+  useTree_unstable,
+  useTreeStyles_unstable,
+  useTreeContextValues_unstable,
+  TreeProvider,
+  TreeSlots,
+} from '@fluentui/react-tree';
+import { FixedSizeList, FixedSizeListProps, ListChildComponentProps } from 'react-window';
+import { ForwardRefComponent, getSlots } from '@fluentui/react-components';
+
+const item1: FlatTreeItemProps[] = [
+  {
+    id: 'flatTreeItem_lvl-1_item-1',
+    children: <TreeItemLayout>Level 1, item 1</TreeItemLayout>,
+  },
+  ...new Array(300).fill(undefined).map<FlatTreeItemProps>((_, i) => ({
+    id: `flatTreeItem_lvl-1_item-1--child:${i}`,
+    parentId: 'flatTreeItem_lvl-1_item-1',
+    children: <TreeItemLayout>Item {i + 1}</TreeItemLayout>,
+  })),
+];
+const item2: FlatTreeItemProps[] = [
+  {
+    id: 'flatTreeItem_lvl-1_item-2',
+    children: <TreeItemLayout>Level 1, item 2</TreeItemLayout>,
+  },
+  ...new Array(300).fill(undefined).map<FlatTreeItemProps>((_, i) => ({
+    id: `flatTreeItem_lvl-1_item-2--child:${i}`,
+    parentId: 'flatTreeItem_lvl-1_item-2',
+    children: <TreeItemLayout>Item {i + 1}</TreeItemLayout>,
+  })),
+];
+
+type FixedSizeTreeProps = Omit<TreeProps, 'children'> & {
+  listProps: FixedSizeListProps;
+};
+
+const FixedSizeTree: ForwardRefComponent<FixedSizeTreeProps> = React.forwardRef((props, ref) => {
+  const state = useTree_unstable(props, ref);
+  useTreeStyles_unstable(state);
+  const contextValues = useTreeContextValues_unstable(state);
+  const { slots, slotProps } = getSlots<TreeSlots>(state);
+  return (
+    <TreeProvider value={contextValues.tree}>
+      <slots.root {...slotProps.root}>
+        <FixedSizeList {...props.listProps} />
+      </slots.root>
+    </TreeProvider>
+  );
+});
+
+const FixedSizeTreeItem = (props: ListChildComponentProps) => {
```

Comment:
nit: might be nicer to extend `ListChildComponentProps` instead of casting like in

https://github.com/microsoft/fluentui/blob/7bf995b3b64f3ffb07291fc6f7efedb00611f2de/packages/react-components/react-table/stories/Table/Virtualization.stories.tsx#L57-L59